workspaces/server/src/infrastructure/repository/cloudflare/user.ts

```diff
@@ -217,12 +219,24 @@ export class CloudflareUserRepository implements IUserRepository {
 			updatedAt: new Date(),
 		};
 
-		const res = await this.client
-			.update(schema.userProfiles)
-			.set(value)
-			.where(eq(schema.userProfiles.userId, userId));
-
-		if (!res.success) {
+		try {
+			await this.client
+				.update(schema.userProfiles)
+				.set(value)
+				.where(eq(schema.userProfiles.userId, userId));
+
+			await this.client
+				.delete(schema.socialLinks)
+				.where(eq(schema.socialLinks.userId, userId));
+
+			await this.client.insert(schema.socialLinks).values(
+				// TODO: 型をちゃんとする
+				(payload.socialLinks ?? []).map((link) => ({
+					userId,
+					url: link,
+				})),
+			);
```

Comment:
そもそもsocialLinksなければクエリする必要はないからこうかな？
```suggestion
      if (payload.socialLinks) {
			  await this.client.insert(schema.socialLinks).values(
				  payload.socialLinks.map((link) => ({
					  userId,
					  url: link,
				  })),
			  );
			}
```